packaging_automation/common_tool_methods.py

```diff
@@ -0,0 +1,286 @@
+    project_version_details = get_version_details(version)
+    return f'{project_version_details["major"]}.{int(project_version_details["minor"]) + 1}'
+
+
+def get_last_commit_message(path: str) -> str:
+    repo = get_new_repo(path)
+    commit = repo.head.commit
+    return commit.message
+
+
+def is_major_release(version: is_version(str)) -> bool:
+    version_info = get_version_details(version)
+    return version_info["patch"] == "0"
+
+
+def str_array_to_str(str_array: List[str]) -> str:
+    return f"{os.linesep.join(str_array)}{os.linesep}"
+
+
+def get_prs_for_patch_release(repo: Repository.Repository, earliest_date: datetime, base_branch: str,
+                              last_date: datetime = None):
+    pull_requests = repo.get_pulls(state="closed", base=base_branch, sort="created", direction="desc")
+
+    # filter pull requests according to given time interval
+    filtered_pull_requests = list()
+    for pull_request in pull_requests:
+        if not pull_request.is_merged():
+            continue
+        if pull_request.merged_at < earliest_date:
+            continue
+        if last_date and pull_request.merged_at > last_date:
+            continue
+
+        filtered_pull_requests.append(pull_request)
+
+    # finally, sort the pr's by their merge date
+    sorted_pull_requests = sorted(filtered_pull_requests, key=lambda p: p.merged_at)
+    return sorted_pull_requests
+
+
+def filter_prs_by_label(prs: List[PullRequest.PullRequest], label_name: str):
+    filtered_prs = []
+    for pr in prs:
+        if any(label.name == label_name for label in pr.labels):
+            filtered_prs.append(pr)
+    return filtered_prs
+
+
+def file_includes_line(base_path: str, relative_file_path: str, line_content: str) -> bool:
+    with open(f"{base_path}/{relative_file_path}", "r") as reader:
+        content = reader.read()
+        lines = content.splitlines()
+        for line in lines:
+            if line == line_content:
+                return True
+    return False
+
+
+def count_line_in_file(base_path: str, relative_file_path: str, search_line: str) -> int:
+    with open(f"{base_path}/{relative_file_path}", "r") as reader:
+        content = reader.read()
+        lines = content.splitlines()
+    return len(list(filter(lambda line: line == search_line, lines)))
+
+
+def replace_line_in_file(file: str, match_regex: str, replace_str: str) -> bool:
+    with open(file, "r") as reader:
+        file_content = reader.read()
+        lines = file_content.splitlines()
+        has_match = False
+        for line_number, line in enumerate(lines):
+            if re.match(match_regex, line.strip()):
+                has_match = True
+                lines[line_number] = replace_str
+        edited_content = str_array_to_str(lines)
+    with open(file, "w") as writer:
+        writer.write(edited_content)
+
+    return has_match
+
+
+def append_line_in_file(file: str, match_regex: str, append_str: str) -> bool:
```

Comment:
I couldn't quite understand what we are exactly doing in `append_line_in_file` & `prepend_line_in_file`.

Can you improve the readability of those two functions, or at least let's comment why do we do `anIndex +1` / `anIndex +2` kind of things.

Reply:
Fixed